### PR TITLE
E2e rx character count

### PR DESCRIPTION
**Pull request by Droogans**
@glynnis I was really happy to see that you wrote out end to end tests for your new component. Most devs don't do that :cry:.

However, there were some missing parts. I've gone ahead and addressed them one at a time, leaving a detailed commit messages overviewing each change so that you know what an ideal page object looks like.

I'll squash all these together once you get the gist of what's going on here, and you're good with merging it.

### CODE REVIEW

File: src/rxCharacterCount/rxCharacterCount.exercise.js [hunk 1]
```diff
@@ -26,11 +38,93 @@ exports.rxCharacterCount = function (options) {
...
+
+        it('should not set the over-limit class on an empty text box', function () {
+            expect(component.isOverLimit()).to.eventually.be.false;
+        });
+
+        var belowNearLimitLength = options.maxCharacters + 1 - options.lowerBoundary;
+        it('should set the near-limit class when ' + belowNearLimitLength + ' characters are entered', function () {
+            component.comment = Array(belowNearLimitLength).join('a');
+            expect(component.isNearLimit()).to.eventually.be.false;
+        });
```
Comment (Droogans) [inline src/rxCharacterCount/rxCharacterCount.exercise.js]: It should *not* set the near-limit class.

File: src/rxCharacterCount/rxCharacterCount.page.js [hunk 2]
```diff
@@ -35,27 +51,40 @@ var rxCharacterCount = {
...
 
+    /**
+       @returns {Number} The remaining number of characters that can be entered.
+     */
     remaining: {
         get: function () {
-            return this.lblRemaining.getText();
+            return this.lblRemaining.getText().then(function (remainingCharacters) {
+                return parseInt(remainingCharacters, 10);
+            });
```
Comment (Droogans) [inline src/rxCharacterCount/rxCharacterCount.page.js]:
```js
return this.lblRemaining.getText().then(parseInt);
```

File: src/rxCharacterCount/rxCharacterCount.exercise.js [hunk 3]
```diff
@@ -17,7 +25,11 @@ exports.rxCharacterCount = function (options) {
         options = {};
     }
 
-    options = _.defaults(options, { /* defaults go here */ });
+    options = _.defaults(options, {
+        maxCharacters: 254,
+        lowerBoundary: 10,
+        ignoreInsignificantWhitespace: true
+    });
```
Comment (Droogans) [inline src/rxCharacterCount/rxCharacterCount.exercise.js]: `lowerBoundary` isn't very intuitive. Pick something like `nearLimitCharacters` or something instead.

**Comment by parlarjb** [comment]
Since you're already bundling some exercise changes here, could you also make sure to add the exercise file to this? https://github.com/rackerlabs/encore-ui/blob/master/grunt-tasks/component-template/rename.json Right now, the exercise file doesn't get named properly when running `grunt-init`

**Comment by Droogans** [comment]
@parlarjb got it.

**Pull request merged**